### DISCUSSION TITLE
See light transition changes live

### DISCUSSION THREAD

**Discussion by SaturnusDJ** [Koenkk/zigbee2mqtt#9618] [General]
This probably is asked before but I cannot find it.

2 years ago when still running the Philips Hue Hub, when a light was doing a transition, this could be followed in HA. E.g. the brightness slider was moving while looking at it. Current situation is that it on screen instantly jumps to the end point of the transition, while the transition is still busy, resulting in a difference between shown digital values and actual real world brightness / color temp.

Are live changes possible in Z2M?

**Comment by SaturnusDJ** [comment]
@Koenkk 
Can you say something about this please?

**Reply by Koenkk** [reply]
The proper way to do this is via reporting (a native feature which most devices should support). You can set it up via the frontend, click on your device -> reporting and set it up for cluster `genLevelCtrl`, attribute `currentLevel`, for change you could fill in `0`, min `0` and `max` 1000.

An overview of which devices support reporting can be found here: https://www.zigbee2mqtt.io/guide/usage/binding.html#state-changes

**Reply by sjorge** [reply]
If you do this, you might need to filter out 'brightness' for optimistic as wel, as otherwise it flaps to whatever you set until the first report arrives.

**Reply by SaturnusDJ** [reply]
Thanks guys! Got some lights working by adding a `LevelCtrl` cluster as Koenkk said. Will do the optimistic filter soon. I guess this reporting config via frontend is being kept somewhere when doing Docker container is upgraded? It cannot be put in the regular configuration.yaml?

For the following devices I cannot get the `LevelCtrl` set up.
https://www.zigbee2mqtt.io/devices/9290012573A.html
https://www.zigbee2mqtt.io/devices/915005106701.html
https://www.zigbee2mqtt.io/devices/8718696449691.html

```
error Request 'zigbee2mqtt/bridge/request/device/configure_reporting' failed with error: 'ConfigureReporting 0x************/11 genLevelCtrl([{"attribute":"currentLevel","minimumReportInterval":"0","maximumReportInterval":3600,"reportableChange":0}], {"sendWhenActive":false,"timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Timeout - 46686 - 11 - 30 - 8 - 7 after 10000ms)'
```

I am not sure if I understand the documentation correctly. I made a group and added 9290012573A to it, but then too there is no transition on the screen. Is a binded device mandatory to get it working for the above three devices?

**Reply by sjorge** [reply]
Yes, the info is kept in the database. Well just zo z2m knows about it. The actually config is kept on the device itself.

Older pre BT hue does indeed not support ANY reporting, I avoided them for that reason, but luckily the with BT bulbs all seem to support reporting aside for colorMode which you probably don’t need.

**Reply by SaturnusDJ** [reply]
Playing around with this on 9290022169. Seems like cluster `LColorCtrl`, attribute `colorTemperature` does color temperature. The downside though, when this one is activated, there is a message every 10 seconds. In addition, to disable/remove this, the light has to be factory reset. Changing parameters (seems to have little effect) or using the disable button won't stop the messages.

**Reply by sjorge** [reply]
It depends on how you configure it, I usually use min=3, max=3600, change=1

And this only sends updates on changes at a max rate of 1 every 3 seconds, otherwise it does a minimal of 1 per hour.

**Reply by SaturnusDJ** [reply]
That minimal value seems like a good idea. Takes away from the smoothness of a moving slider when a quick transition is running, but works more stable and also better in Apple Home where with min=1 there are set vs. get conflicts.
Overall, I might use reporting for lights that won't throw out updates every 10 seconds despite the values used for reporting. But that is not gonna be a lot of lights. Mostly got 'old' Philips lights here. 🙇‍♂️
